### PR TITLE
CLDR-16846 xlsx to xml converter for bulk upload

### PR DESCRIPTION
**Pull request by srl295**
- move bulk upload jsp link into a new 'upload' panel
- add a converter from spreadsheet (xlsx/csv/etc)  (the Dashboard Download) to bulk-upload XML
- CLDRFile: add a 'leniency' option, disable validation for XML upload (it gets triple checked anyway)

CLDR-16846

- [ ] This PR completes the ticket.

FYI, draft

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-apps/js/src/esm/cldrBulkConverter.mjs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        <types/>
+    </localeDisplayNames>
+    <units>
+        <unitLength type="long"/>
+    </units>
+    <annotations/>
+    <characters/>
+    <numbers>
+        <miscPatterns numberSystem="latn"/>
+    </numbers>
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrBulkConverter.mjs:97]: realized this skeleton needs to be here to keep the XML in order… almost 100% from client!

**Comment by macchiati** [comment]
looks good to me. Still draft?

**Comment by srl295** [comment]
Thanks. Yes. It doesn’t have Dtd ordering so fails.

File: tools/cldr-apps/js/src/esm/cldrBulkConverter.mjs (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/js/src/esm/cldrBulkConverter.mjs]: fixed w/ leniency on cldrfile

**Comment by srl295** [comment]
@macchiati ptal, i'm posting it to staging now

UI: 
image

image_720


image

**Comment by srl295** [comment]
image
 updated wording slightly

**Review by macchiati — APPROVED** [review]

**Comment by macchiati** [comment]
BTW, I think it would have been easier to adjust the bulk upload to take a <xpath, value> TSV file as well as an XML file. That would be useful for many other reasons, and we wouldn't have to have the complications of generating compliant XML in javascript (eg the ordering of items etc.)

**Comment by srl295** [comment]
> BTW, I think it would have been easier to adjust the bulk upload to take a <xpath, value> TSV file as well as an XML file. That would be useful for many other reasons, and we wouldn't have to have the complications of generating compliant XML in javascript (eg the ordering of items etc.)

That could be a way to do that, except that the bulk upload is a JSP, was trying to not modify that (it needs a full modernize). Didn't end up having to do anything for ordering other than change the bulk upload to not validate XML, which will help anyone else doing a bulk upload also.

This way in the future we can make the bulk upload interactive - the client side can (via voting api) check each item and verify what you want to do with it.  When you throw the whole file over to the server, then it's harder to make a UI to work with certain parts.

**Pull request merged**